api/namex/services/cloud_event_messages/cloud_event_messages.py

```diff
@@ -0,0 +1,69 @@
+# Copyright © 2021 Province of British Columbia
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import asyncio
+from datetime import datetime, timedelta, timezone
+import json
+import time
+import uuid
+import nats
+from typing import Optional
+from sentry_sdk import capture_message
+
+from queue_common.messages import create_cloud_event_msg
+from queue_common.service import QueueServiceManager
+from queue_common.service_utils import QueueException, logger
+from config import Config
+
+
+class CloudEventMessageService(object):
+    """Provides services to cloud event message services"""
+    
+    @staticmethod
+    async def sendNameRequestStateEvent(nr_num, old_nr_num, state_cd, old_state_cd):
+        event_loop = asyncio.get_event_loop()
+        qsm = QueueServiceManager()  # pylint: disable=invalid-name
+        await qsm.run(loop=event_loop,
+                      config=Config,
+                      callback=cb_subscription_handler)
+        cloud_event_msg = create_cloud_event_msg(msg_id=str(uuid.uuid4()),
+                                                msg_type='bc.registry.names.request',
+                                                source=f'/requests/{nr_num}',
+                                                time=datetime.
+                                                utcfromtimestamp(time.time()).
+                                                replace(tzinfo=timezone.utc).
+                                                isoformat(),
+                                                identifier=nr_num,
+                                                json_data_body={
+                                                    'request': {
+                                                        'nrNum': nr_num,
+                                                        'oldNrNum': old_nr_num,
+                                                        'newState': state_cd,
+                                                        'previousState': old_state_cd
```

Comment:
Included new and old values as @sumesh-aot pointed out that the update from NRL to NR may be necessary:
https://github.com/bcgov/entity/pull/8003/files#r660844340

Reply:
Do you store the account-id anywhere in names? To auto-affiliate the NR to the account, we will need to know the account to affiliate.

Reply:
Do you mean the account-id from the user who created the NR?

Reply:
Yes, if it's an authenticated user.

Reply:
Another option for us is to get the account information from Pay records for the NR.

Reply:
I see there's a user_id column for the `requests` and `events` tables but they are not populating with the user_id from the user who is creating the NR in namex database.

Reply:
it is if the user is logged in.

Reply:
Yes, it is possible to get the logged user from the request but namex-api is not saving the logged user. It saves name request account service userId in `requests` and `events` tables.

Reply:
ex. non-authenticated user submitted NR
```
   nr_num   |    state_cd     |           username
------------+-----------------+------------------------------
 NR 3640335 | DRAFT           | name_request_service_account
 NR 3640335 | PENDING_PAYMENT | name_request_service_account
 NR 3640335 | PENDING_PAYMENT | name_request_service_account
 NR 3640335 | DRAFT           | name_request_service_account
 NR 3640335 | INPROGRESS      | idir\kyee
 NR 3640335 | INPROGRESS      | idir\kyee
 NR 3640335 | INPROGRESS      | idir\kyee
 NR 3640335 | APPROVED        | idir\kyee
 NR 3640335 | APPROVED        | nro_service_account
 ```
 Looks like non staff are not captured via namerequest ... wtf

Reply:
in any case, first emitted message is in the DRAFT state, which is after payment.
so currently, PAY never has any NRLs, so there is no oldNrNum that is used.

Reply:
Yes, I fixed that. Only NRLs changes are not sending messages.